lib/uri/common.rb

```diff
@@ -34,7 +34,7 @@ def make_components_hash(klass, array_hash)
       tmp = {}
       if array_hash.kind_of?(Array) &&
           array_hash.size == klass.component.size - 1
-        klass.component[1..-1].each_index do |i|
+        0.upto(klass.component.size - 2) do |i|
```

Comment:
You could use ``(klass.component.size - 1).times do |i|``.

Reply:
It has to be `klass.component[i + 1]` because the first item of`klass.component` is `scheme` and it's not included in `array_hash`.

Reply:
I applied `array_hash.size.times do |i|`. Thanks!

Reply:
But you could replace ``klass.component[i + 1]`` with ``klass.component[i]`` when ``i`` starts at 1, using my second edit.
This would avoid doing plus one operations, probably a lit bit faster for many elements.

Reply:
In that case, `array_hash[i - 1]` needs to be used.

Reply:
Yes, just noticed that ``i`` is used in both sides of assignment, sorry.

Maybe using ``array_hash.each_with_index {|value,i|`` and replacing ``array_hash[i]`` for ``value`` could also improve performance and legibility.

Reply:
Nope, `each_with_index` is slower.